SalesforceCommerceCloud/pwa-kit
# Pass in 'headers' and 'rawResponse' options to mutation hooks

Like the query hooks in `commerce-sdk-react`, we want the mutation hooks to also accept `headers` and `rawResponse` options and pass them down to the underlying isomorphic sdk

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- User can pass in `headers` and `rawResponse` values when passing the desired API endpoint name

# How to Test-Drive This PR

- In line 58 of `test-commerce-sdk-react>app>pages>use-shopper-orders.tsx`, pass in desired `headers` and `rawResponse` values (make sure to have a valid `basketId`)
```
hook: useShopperOrdersMutation({action, headers: {'test-header': 'value'}, rawResponse: false})
```
- Run `npm start` in `test-commerce-sdk-react`
- Go to shopperOrders page
- Click on `createOrder`
- Verify that the correct response body is being sent back dependent on the `rawResponse` value that is set by the user

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
